src/svm/impls/binary/binary.c

```diff
@@ -1549,6 +1708,73 @@ PetscErrorCode SVMGridSearch_Binary(SVM svm)
   PetscFunctionReturn(0);
 }
 
+#undef __FUNCT__
+#define __FUNCT__ "SVMLoadMatGramian_Binary"
+PetscErrorCode SVMLoadMatGramian_Binary(SVM svm,PetscViewer v)
+{
+  MPI_Comm comm;
+  Mat      G;
+
+  PetscFunctionBegin;
+  TRY( PetscObjectGetComm((PetscObject) svm,&comm) );
+
+  /* Create matrix */
+  TRY( MatCreate(comm,&G) );
+  TRY( MatSetType(G,MATDENSE) );
+  TRY( PetscObjectSetName((PetscObject) G,"G") );
+  TRY( PetscObjectSetOptionsPrefix((PetscObject) G,"gramian_") );
+  TRY( MatSetFromOptions(G) );
+
+  TRY( MatLoad(G,v) );
+  TRY( SVMSetMatGramian(svm,G) );
+
+  /* Free memory */
+  TRY( MatDestroy(&G) );
+  PetscFunctionReturn(0);
+}
+
+#undef __FUNCT__
+#define __FUNCT__ "SVMViewMatGramian_Binary"
+PetscErrorCode SVMViewMatGramian_Binary(SVM svm,PetscViewer v)
+{
+  MPI_Comm   comm;
+  const char *type_name = NULL;
+
+  Mat        G;
+  PetscInt   M,N;
+
+  PetscBool  isascii;
+
+  PetscFunctionBegin;
+  TRY( SVMGetMatGramian(svm,&G) );
+  if (!G) {
+    TRY( PetscObjectGetComm((PetscObject) v,&comm) );
```

Comment:
use
```
PetscObjectComm((PetscObject) v)
```
in SETERRQ unless comm is available from previous code

Reply:
Accepted.